cocotb/share/lib/vhpi/VhpiCbHdl.cpp

```diff
@@ -99,8 +100,8 @@ bool get_range(vhpiHandleT hdl, vhpiIntT dim, int *left, int *right) {
                 if (curr_idx == dim) {
 
                     vhpi_release_handle(it);
-                    vhpiIntT l_rng = vhpi_get(vhpiLeftBoundP, constraint);
-                    vhpiIntT r_rng = vhpi_get(vhpiRightBoundP, constraint);
+                    int l_rng = static_cast<int>(vhpi_get(vhpiLeftBoundP, constraint));
+                    int r_rng = static_cast<int>(vhpi_get(vhpiRightBoundP, constraint));
 #ifdef IUS
                     if (l_rng != UNCONSTRAINED && r_rng != UNCONSTRAINED) {
```

Comment:
I don't think you should be casting before comparing to `UNCONSTRAINED`.

Can you put the cast on line 111 and 112 instead?